python/slurmTools.py

```diff
@@ -0,0 +1,430 @@
+        dico["#EXEC_PATH#"] = self.execPath
+        dico["#CMS_PATH#"] = CMSSW_BASE
+        dico["#JOB_NAME#"] = "cp3-llbb"
+
+        for sample in self.sampleCfg:
+            name = sample["db_name"]
+            jobList = sample["jobs"]
+
+            for job in jobList:
+
+                jsonFileName = os.path.join(self.inDir, "samples_{}.json".format(jobCount) )
+                jsonContent = copy.deepcopy(sample["json_skeleton"])
+                
+                if "#DB_NAME#" in jsonContent.keys():
+                    jsonContent = {}
+                    jsonContent[name] = copy.deepcopy(sample["json_skeleton"]["#DB_NAME#"])
+                    jsonContent[name]["db_name"] = name
+                
+                jsonContent[name]["files"] = job[0]
+                jsonContent[name]["event-start"] = job[1]
+                jsonContent[name]["event-end"] = job[2]
+                
+	        jsonContent[name]["output_name"] = name + "_histos_{}".format(jobCount)
+
+                with open(jsonFileName, "w") as js:
+                    json.dump(jsonContent, js)
+
+                jobCount += 1
+           
+        dico["#N_JOBS#"] = str(jobCount - 1)
+
+        # Extra sbatch commands
+        extra_commands = ""
+        if self.requestedMemory:
+            extra_commands += "#SBATCH --mem-per-cpu {:.0f}\n".format(self.requestedMemory)
+        if self.runtime:
+            extra_commands += "#SBATCH --time {}\n".format(self.runtime)
+
+        dico["#ADDITIONAL_SBATCH_COMMANDS#"] = extra_commands
+
+        slurmFile = str(self.baseShell)
+
+        for key in dico.items():
+            slurmFile = slurmFile.replace(key[0], key[1])
+
+        # Shell wrapper
+        shFileName = os.path.join(self.inDir, "slurm.sh")
+        with open(shFileName, "w") as sh:
+            sh.write(slurmFile)
+        perm = os.stat(shFileName)
+        os.chmod(shFileName, perm.st_mode | stat.S_IEXEC)
+
+        print "Created {} job command files. Caution: the jobs are not submitted yet!.".format(jobCount)
+
+        self.jobCount = jobCount
+        self.isCreated = True
+
+        # Generate a command to hadd all the files when the jobs are done
+        haddFile = os.path.join(self.baseDir, "hadd_histos.sh")
+        with open(haddFile, "w") as f:
+            cmd = """#!/usr/bin/env bash
+
+function check_success {
+  should_exit=false
+  find #LOGDIR_PATH# -name "*.err" -not -size 0 |
+  {
+    while read f; do
+      [[ $f =~ slurm_([0-9]+) ]]
+      id=${BASH_REMATCH[1]}
+      echo "Error: job #${id} failed."
+      should_exit=true
+    done
+
+    if [ "${should_exit}" == true ]; then
+      exit 1
+    fi
+  }
+
+  if [ $? -eq 1 ]; then
+    exit 1
+  fi
+}
+
+function do_hadd {
+ if [ ! -z "$1" ]; then
+  to_add=( $(ls $1_*.root) )
+  # If there is only one file to hadd, do not do it but simply rename the file (faster)
+  if [[ ${#to_add} == 1 ]]; then
+    mv ${to_add[1]} $1.root
```

Comment:
There's a typo here that I fixed in #142:
- `${#to_add}` should be `${#to_add[*]}`
- `${to_add[1]}` should be `${to_add[0]}` (confusion between bash and zsh array access :p)